resource_obmcs_core_instance.go

```diff
@@ -32,27 +32,38 @@ func InstanceResource() *schema.Resource {
 			"create_vnic_details": {
 				Type:     schema.TypeMap,
 				Optional: true,
+				ForceNew: true,
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
 						"assign_public_ip": {
 							Type:     schema.TypeBool,
 							Optional: true,
+							ForceNew: true,
 						},
 						"display_name": {
 							Type:     schema.TypeString,
 							Optional: true,
+							ForceNew: true,
 						},
 						"hostname_label": {
 							Type:     schema.TypeString,
 							Optional: true,
+							ForceNew: true,
 						},
 						"private_ip": {
 							Type:     schema.TypeString,
 							Optional: true,
+							ForceNew: true,
+						},
+						"skip_source_dest_check": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							ForceNew: true,
```

Comment:
Regarding display_name, hostname_label and skip_source_dest_check having _ForceNew: true_, as these fields can be updated through the vnic api, is there a reasonable expectation from a user that they should be able to update these values here without their instance being recreated? Can they be updated separately while attached to an instance? If so, does that have an ill-effect here when running plan/apply on an existing plan after?

Reply:
Yes, yes, and yes. Basically, there's more work to be done here, but this change at least accurately reflects the current behavior. Currently, any change to these properties on an existing instance will be ignored. Forcing New on these will allow property updates to be applied. The remaining work should be:

1) Update the VNIC for the 3 properties that you mention.
2) Import/Refresh or create_vnic_details.

I'll open items to track these. They shouldn't be difficult, but are a bit strange in that VNIC management will need to happen within the instance resource.

The change in this PR is, I believe, an improvement, just not all the way to where we want to be.